implementation/objectoriented.tex

```diff
@@ -1,9 +1,12 @@
 \section{Object Oriented Embedded Programming}
-\label{sec:back:oo}
+\label{sec:impl:oo}
 
 In its essence, Object Oriented programming focuses on organizing a computer program by looking at the data the program operates on, grouping related data into objects and defining methods that operate on the data contained within the objects.
-The paradigms essential concept can be applied to embedded {\C} programming even though the language itself does not directly define any language features to aid the design.
-This section looks at how Object Oriented programming is used in conjunction with \gls{mmio} in embedded programming.
+In this section we look at how control over the memory layout of objects and static dispatch enables the use of the Object Oriented paradigm in conjunction with \gls{mmio} in embedded programming.
```

Comment:
muligens: "In this section`, <- komma` we look at how control over the memory layout of objects and static dispatch`, can be used to enable` ~~enables the use of~~ the Object Oriented paradigm in conjunction with \gls{mmio} in embedded programming."